bcdaworker/worker/worker.go

```diff
@@ -133,12 +141,23 @@ func (w *worker) ProcessJob(ctx context.Context, job models.Job, jobArgs models.
 			logger.Error("Job failed. Job ID: ", job.ID)
 		}
 	}
+	//move the files over
+	err = moveFiles(tempJobPath, stagingPath)
+	if err != nil {
+		logger.Error(err)
+	}
 
 	err = createJobKeys(ctx, w.r, jobKeys, job.ID)
 	if err != nil {
 		logger.Error(err)
 	}
 
+	//with a job key created, delete the temp job directory.
+	if err = os.RemoveAll(tempJobPath); err != nil {
```

Comment:
Is it possible for this function to return before this is called? Wondering if we want to move this into a `defer` statement right after creating the temp job path.

I could also imagine wanting to create a new temporary subdir for each call to process job, but maybe overkill if the workers process one job at a time.

Reply:
Good call; that latter condition is also a concern I'll address: if a worker is processing multiple keys on the same job, it could remove files that are actually still in-progress for a given job. 

will add a defer block and update the directory structure / scope, thank you!